flink-runtime/src/main/java/org/apache/flink/runtime/zookeeper/ZooKeeperStateHandleStore.java

```diff
@@ -346,17 +346,20 @@ public int exists(String pathInZooKeeper) throws Exception {
 			} else {
 				// Initial cVersion (number of changes to the children of this node)
 				int initialCVersion = stat.getCversion();
-
-				List<String> children = ZKPaths.getSortedChildren(
-						client.getZookeeperClient().getZooKeeper(),
-						ZKPaths.fixForNamespace(client.getNamespace(), "/"));
-
-				for (String path : children) {
-					path = "/" + path;
+				List<String> childrenInStr =
+					client.getZookeeperClient().getZooKeeper().
+						getChildren(ZKPaths.fixForNamespace(client.getNamespace(), "/"), false);
+				List<Long> children = new ArrayList<Long>(childrenInStr.size());
+				for(String childNode : childrenInStr) {
+					children.add(new Long(childNode));
```

Comment:
Ok. I see. I am not sure on this MesosWorker. Using cxid am not sure if we have an API. If so we can direclty use it. Will be back.